kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -254,4 +281,21 @@ private void validate(String value) {
     private static void addConfig(HashMap<String, TableConfig<?>> configs, TableConfig<?> config) {
         configs.put(config.getKey().toLowerCase(Locale.ROOT), config);
     }
+
+    private static Map<String, String> parseStringToMap(Engine engine, String jsonString) {
+        if (jsonString == null) {
+            return Collections.emptyMap();
+        }
+        StructType schema = new StructType()
+                .add("config", new MapType(StringType.STRING, StringType.STRING, true));
+        CloseableIterator<Row> batchRows = engine.getJsonHandler().parseJson(
+                singletonStringColumnVector(String.format("{\"config\": %s}", jsonString)),
+                schema,
+                Optional.empty()
+        ).getRows();
+        assert batchRows.hasNext();
+        Row row = batchRows.next();
+        assert !row.isNullAt(0);
```

Comment:
I think there is a utility method that extracts exactly one element from the iterator. May be in InternalUtils.